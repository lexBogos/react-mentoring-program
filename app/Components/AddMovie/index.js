import React from 'react';
import './index.scss';

const AddMovie = ({onMovieChosen}) => <button className='addMovieButton' onClick={() => {
    onMovieChosen(undefined, 'MOVIE_ADD')
    }
} >+ ADD MOVIE</button>

export default AddMovie;