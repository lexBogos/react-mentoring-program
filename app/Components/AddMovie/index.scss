.addMovieButton{
    position: absolute;
    top: 10%;
    left: 80%;
    font-size:16px;
    font-weight: 400;
    line-height: 45px;
    width:15%;
    background-color: rgba(53, 50, 50, 0.5);
    color: #f65261;
    border: none;
}

.addMovieButton:active {
    box-shadow: 0 5px rgb(65, 61, 61);
    transform: translateY(4px);
  }