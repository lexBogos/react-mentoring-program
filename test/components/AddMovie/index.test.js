import React from 'react';
import AddMovie from '../../../app/Components/AddMovie/index';
import { render } from '@testing-library/react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Enzyme,{ shallow } from 'enzyme';
import '@testing-library/jest-dom/extend-expect';

Enzyme.configure({ adapter: new Adapter() });

describe('AddMovie component', () => {
        it('render AddMovie component toBeInTheDocument example', () => {
            const { getByText } = render(<AddMovie/>);
            const element = getByText('+ ADD MOVIE');

            expect(element).toBeInTheDocument();
        })

        it('render AddMovie component toMatchSnapshot', () => {
            const { asFragment } = render(<AddMovie/>);
            
            expect(asFragment(<AddMovie/>)).toMatchSnapshot();
        })

        it('click on AddMovie button should dispatch event', () => {

                const mockCallBack = jest.fn();
                render(<AddMovie onMovieChosen={mockCallBack}/>);
                const addMovieComponent = shallow(<AddMovie onMovieChosen={mockCallBack}/>);
                addMovieComponent.find('button').simulate('click');
    
                expect(mockCallBack.mock.calls.length).toEqual(1);
        })
})