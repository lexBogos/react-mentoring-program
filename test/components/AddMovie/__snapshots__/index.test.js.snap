// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMovie component render AddMovie component toMatchSnapshot 1`] = `
<DocumentFragment>
  <button
    class="addMovieButton"
  >
    + ADD MOVIE
  </button>
</DocumentFragment>
`;
